parsely/parselyandroid/ParselyTracker.java

```diff
@@ -469,4 +477,38 @@ protected Void doInBackground(Void... params) {
             return null;
         }
     }
+
+    public class GetAdKey extends AsyncTask<Void, Void, String> {
+        private Context mContext;
+
+        public GetAdKey(Context context) {
+            mContext = context;
+        }
+
+        @Override
+        protected String doInBackground(Void... params) {
+            AdvertisingIdClient.Info idInfo = null;
+            String advertId = null;
+            try {
+                idInfo = AdvertisingIdClient.getAdvertisingIdInfo(mContext);
+            } catch (GooglePlayServicesRepairableException |IOException|GooglePlayServicesNotAvailableException e) {
+                PLog("No Google play services or error! falling back to device uuid");
+                // fall back to device uuid on google play errors
+                advertId = getSiteUuid();
```

Comment:
This works without `this` because this class is nested within `ParselyTracker`, right?

Reply:
yup, inner classes can access outer class variables like this (unless I had declared a variable in the inner class with the same name)